src/deepqmc/utils.py

```diff
@@ -102,3 +117,32 @@ def InverseSchedule(init_value, decay_rate):
 def argmax_random_choice(rng, x):
     mask = x == x.max()
     return jax.random.choice(rng, jnp.arange(len(x))[mask])
+
+
+def segment_nanmean(data, segment_ids, num_segments=None):
+    mask = ~jnp.isnan(data)
+    data_zero = jnp.where(mask, data, 0)
+    nanmean = ops.segment_sum(data_zero, segment_ids, num_segments) / mask.sum()
+    return nanmean
+
+
+def segment_nanstd(data, segment_ids, num_segments=None):
+    mask = ~jnp.isnan(data)
+    nanmean = segment_nanmean(data, segment_ids, num_segments)
+    nanstd = jnp.where(mask, (nanmean[segment_ids] - data) ** 2, 0)
+    nanstd = jnp.sqrt(ops.segment_sum(nanstd, segment_ids, num_segments) / mask.sum())
+    return nanstd
+
+
+def per_config_stats(n_configs, data, config_idx, prefix):
+    mean = segment_nanmean(data, config_idx, n_configs)
```

Comment:
What is the return type of `segment_nanmean`? Is it a tensor with dimension equal to the number of configs, or is it an average over all the configs?

Reply:
The interface mirrors the segment operations from the [jax.ops](https://jax.readthedocs.io/en/latest/jax.ops.html) module, i.e. it returns an array with dimension equal to the number of configs.

Reply:
thanks